lib/rubygems.rb

```diff
@@ -798,24 +798,20 @@ def self.open_file(path, flags, &block)
     File.open(path, flags, &block)
   end
 
+  MODE_TO_FLOCK = IO::RDONLY | IO::APPEND | IO::CREAT # :nodoc:
```

Comment:
Now I think it'd be more readable to remove this constant and pass the mode directly to the `File.open` call, given it's the only place using it?

Reply:
@nobu Let me know what you think. I feel the code is easier to follow if the mode used is right in there, but if you have a very strong opinion about using a constant, I don't mind. I'd make it private if we end up using a constant.

Reply:
It is for #7939.
`IO::SHARE_DELETE` is necessary on Windows, but it is not defined in TruffleRuby.